### ISSUE TITLE
Package sending status modal seems to be erroneous / improvement 

### ISSUE THREAD

**Issue by darupp** [hbz/laser-ygor#223]
_GOKb 8.5.2, phaeton-qa; YGOR 0.88, phaeton-dev:_

The modal that appears after package sending seems to be wrong. After testing (EM120, see corresponing issues), the message is:

> GOKb Antwort
> Intakt: 0, Fehlerhaft: 0 

... although the package and all 184 tipps have been successfully imported, as a successive RESTer API call with the same package JSON shows: 
```
{
    "results": [],
    "result": "OK",
    "message": "Created/Updated package Emerald EM120 Testpaket 2020-03-05 with 184 TIPPs. (Previously: 184, Newly Retired/Deleted: 0)",
    "pkgId": 6339430,
    "uuid": "18863088-06db-4be3-b5d4-ea5e8aa3444e",
    "errors": []
}
```

(There could be still a problem in the uploading process - the response is from RESTer)

# Improved response modal

The Answer should not only include parts of the GOKb response: (at least until we have dry run functionality)

  * Total some of accepted and rejected titles (as intended)
  * Package error message like "Package was created, but 1 TIPPs could not be created!"
  * Two column table with rejected titles and reason for rejection: Titlename - Error Message like "TIPP Validation for title Handbook of business strategy failed: [Coverage end date must not be prior to its start date!]"

# Example GOKb Response

```
{
    "results": [],
    "result": "ERROR",
    "message": "Package Emerald 120 Testpaket 2020-03-05 was created, but tipps have not been loaded because of validation errors!",
    "errors": [
        {
            "code": 400,
            "idx": 133,
            "message": "TIPP Validation for title Handbook of business strategy failed: [Coverage end date must not be prior to its start date!]",
            "baddata": {
                "coverage": [
                    {
                        "coverageDepth": "Fulltext",
                        "startIssue": "1",
                        "endDate": "2001",
                        "endVolume": "7",
                        "startVolume": "5",
                        "endIssue": "1",
                        "startDate": "2004"
                    }
                ],
                "﻿publication_title": "Handbook of Business Strategy",
                "package": {
                    "internalId": 6328792
                },
                "title": {
                    "internalId": 489723,
                    "identifiers": [
                        {
                            "type": "onlineIdentifier",
                            "value": "2054-1775"
                        },
                        {
                            "type": "printIdentifier",
                            "value": "1077-5730"
                        },
                        {
                            "type": "emerald",
                            "value": "hbs"
                        },
                        {
                            "type": "zdb",
                            "value": "2228293-2"
                        }
                    ],
                    "name": "Handbook of business strategy",
                    "type": "Serial"
                },
                "platform": {
                    "internalId": 64378,
                    "name": "Emerald Insight",
                    "primaryUrl": "https://www.emerald.com/insight"
                },
                "url": "https://www.emerald.com/insight/publication/issn/1077-5730"
            },
            "errors": [
                "Coverage end date must not be prior to its start date!"
            ]
        }
    ]
}
```

**Comment by philboeselager** [comment]
@darupp This is a refinement of #197, right? I would copy paste the refinements there then.

**Comment by darupp** [comment]
The error response will be improved in four steps:
1. Using current error response; static dropdown panel with package error message and table with tipp error messages; sorted by title name
2. Better title identification via ygor title number and idx field
3. Improvement of gokb response error object and working with error codes, i18n for internationalized error messages
4. dryrun functionality in GOKb and direct error editing in ygor tables

**Comment by croqueGrec09** [comment]
Tested with Emerald 120 package, all three enrichment status checked. Seems to work properly; no unrealistic jumps are perceivable.

**Comment by croqueGrec09** [comment]
Package sending to GOKb is a bit weird; when I submit my credentials, I do not get any feedback whether the process is started or not. I can log in twice or as many times I wish; the process is triggered only once (according to job ist in GOKb). Feedback whether the job is running would be very useful.

**Comment by philboeselager** [comment]
> when I submit my credentials, I do not get any feedback whether the process is started or not.

Feedback is given by the existence of a progress bar. As long as there is a progress bar, the upload is taking place.

> I can log in twice or as many times I wish; the process is triggered only once (according to job ist in GOKb).

Is now prevented by disabling upload buttons (see [last commit](https://github.com/hbz/laser-ygor/commit/79f00a36911473c721d951483c0e3e7b6095460d)). Upload buttons do get enabled again when something is changed on the data (like a fix in a title).

> Feedback whether the job is running would be very useful.

Also, the displaying of a results table is fixed.

> Also the GOKb response button is still dead (as it was before triggering the sending job)

I'm not sure what is meant by this. @croqueGrec09 Please check by re-testing and comment if something is still missing.

**Comment by croqueGrec09** [comment]
I mean the button on top /statistic/index, entitled GOKb response. This button does nothing, no matter, whether I sent the package to GOKb or not.

**Comment by philboeselager** [comment]
> I mean the button on top /statistic/index, entitled GOKb response. This button does nothing, no matter, whether I sent the package to GOKb or not.

I understand. This "button" was implemented as a toggle in a previous version, before there was the green button alongside that removes the GOKb response entirely.

By now, this (former) button is meant as an optical separator or a kind of headline. But I can re-activate the toggle function, though this was classified as a "nice to have" and therefore has been lowered in priority. Does this appear more than a nice-to-have to you, @croqueGrec09 ?

I'm going to talk to @MiriKon about it, since she is the "professional" instructor by now.

**Comment by croqueGrec09** [comment]
Alright, the thing is that after sending package to GOKb, I see nothing (with v0.91) from GOKb. I have no progress bar, no list, nothing, no signal on behalf of GOKb. That is very irritating for me but it may be that I lack something, @MiriKon ...

**Comment by philboeselager** [comment]
This is a bit weird. I'm getting fine results on the test server. Can you please check with a small package and another browser, @croqueGrec09 ?

**Comment by philboeselager** [comment]
Done.
